weed/storage/volume_read_write.go

```diff
@@ -117,7 +122,66 @@ func (v *Volume) writeNeedle(n *needle.Needle, fsync bool) (offset uint64, size
 	return
 }
 
-func (v *Volume) deleteNeedle(n *needle.Needle) (uint32, error) {
+func (v *Volume) writeNeedle2(n *needle.Needle, fsync bool) (offset uint64, size uint32, isUnchanged bool, err error) {
```

Comment:
parameter fsync is not used?

Reply:
yes. In current implement, we don't need to fsync each file. just need to fsync in each submit which may contain serval files.   we can also add a option for fsync, but it is better to add this option in collection instead file， just use fsync in data sensitive collection.

Reply:
> So with this patch, the "fsync" is not doing fsync for every request?

yes. this patch use async request, so we can do fsync after collect several requests. but when user get a response, the write data has been flushed to disk. it is reliable for user.

Reply:
if not used, you can remove it.

Reply:
Add fsync option to keep compatible with before. 

1.benchmark without fsync:
./weed benchmark -master=localhost:9333 -c=1024 -n=1000000 -size=10000

Concurrency Level:      1024
Time taken for tests:   46.972 seconds
Complete requests:      1000000
Failed requests:        0
Total transferred:      10031507488 bytes
Requests per second:    21289.11 [#/sec]
Transfer rate:          208556.55 [Kbytes/sec]

Connection Times (ms)
              min      avg        max      std
Total:        0.5      47.5       179.0      14.8

Percentage of the requests served within a certain time (ms)
   50%     46.7 ms
   66%     50.3 ms
   75%     54.4 ms
   80%     58.9 ms
   90%     66.7 ms
   95%     72.6 ms
   98%     84.0 ms
   99%     91.6 ms
  100%    179.0 ms

2.benchmark with fsync:
./weed benchmark -master=localhost:9333 -c=1024 -n=100000 -size=10000 -fsync=true

Concurrency Level:      1024
Time taken for tests:   14.322 seconds
Complete requests:      100000
Failed requests:        0
Total transferred:      1003135452 bytes
Requests per second:    6982.30 [#/sec]
Transfer rate:          68400.34 [Kbytes/sec]

Connection Times (ms)
              min      avg        max      std
Total:        3.0      145.3       420.3      72.8

Percentage of the requests served within a certain time (ms)
   50%    138.7 ms
   66%    176.5 ms
   75%    198.3 ms
   80%    211.2 ms
   90%    245.4 ms
   95%    270.6 ms
   98%    303.3 ms
   99%    323.9 ms
  100%    420.3 ms

If fsync is not set, the performance is as same as before, if fsync is set, we get more than 5X performance to beforece.